staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -119,20 +119,43 @@ type DRAPlugin interface {
 	// must have an entry in the response, even if that entry is nil.
 	UnprepareResourceClaims(ctx context.Context, claims []NamespacedObject) (result map[types.UID]error, err error)
 
-	// ErrorHandler gets called for each error encountered while publishing
-	// ResourceSlices. See [resourceslice.Options.ErrorHandler] for details.
+	// HandleError gets called for errors encountered in the background,
+	// for example while publishing ResourceSlices.
+	// See [resourceslice.Options.ErrorHandler] for details on that.
 	//
-	// A simple implementation is to only log with k8s.io/apimachinery/pkg/util/runtime.HandleErrorWithContext:
-	//    runtime.HandleErrorWithContext(ctx, err, msg)
+	// The recommended implementation is to log with
+	// runtime.HandleErrorWithContext(ctx, err, msg) (from
+	// k8s.io/apimachinery/pkg/util/runtime.HandleErrorWithContext)
+	// and then to exit the process if the error is fatal.
+	// Ideally the process should shut down gracefully, which can be
+	// achieved by canceling the main context of the DRA driver.
+	//
+	// Fatal errors can be distinguished from recoverable errors via
+	//    errors.Is(err, kubeletplugin.ErrRecoverable)
 	//
 	// This is a mandatory method because drivers should check for errors
 	// which won't get resolved by retrying and then fail or change the
 	// slices that they are trying to publish:
 	// - dropped fields (see [resourceslice.DroppedFieldsError])
 	// - validation errors (see [apierrors.IsInvalid])
-	ErrorHandler(ctx context.Context, err error, msg string)
+	HandleError(ctx context.Context, err error, msg string)
+}
+
+// ErrRecoverable distinguishes recoverable errors from those errors which are fatal
+// and should cause the process to exit. Use with:
+//
+//	errors.Is(err, ErrRecoverable)
+var ErrRecoverable = errors.New("recoverable error")
```

Comment:
Should we invert this to be `ErrIrrecoverable` if it positively indicates a fatal error?

Reply:
Oh I think I misread this.